Flow on Windows fails with "Error looking up function 'write': The specified procedure could not be found"

### Describe the issue

Receive an error executing the following flow on 0.16.6 on windows standalone

```
id: test
namespace: blueprint

tasks:
  - id: export
    type: io.kestra.plugin.jdbc.postgresql.Query
    url: jdbc:postgresql://postgres_server:5432/postgres
    username: admin
    password: pw
    store: true
    sql: SELECT * FROM test LIMIT 10

  - id: write_to_csv
    type: io.kestra.plugin.serdes.csv.CsvWriter
    from: "{{ outputs.export.uri }}"

  - id: process_output
    type: io.kestra.plugin.scripts.python.Script
    docker:
      image: ghcr.io/kestra-io/pydata:latest
    script: |
      print("{{ outputs.export.uri }}")

```

```
Exception in thread "docker-java-stream-930316280" java.lang.NoClassDefFoundError: Could not initialize class com.github.dockerjava.transport.DomainSocket
        at com.github.dockerjava.transport.UnixSocket.get(UnixSocket.java:29)
        at com.github.dockerjava.zerodep.ApacheDockerHttpClientImpl$2.createSocket(ApacheDockerHttpClientImpl.java:154)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:125)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.io.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:409)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:164)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalExecRuntime.connectEndpoint(InternalExecRuntime.java:174)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ConnectExec.execute(ConnectExec.java:135)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement$1.proceed(ExecChainElement.java:57)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ProtocolExec.execute(ProtocolExec.java:172)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement$1.proceed(ExecChainElement.java:57)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.HttpRequestRetryExec.execute(HttpRequestRetryExec.java:93)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement$1.proceed(ExecChainElement.java:57)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ContentCompressionExec.execute(ContentCompressionExec.java:128)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement$1.proceed(ExecChainElement.java:57)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.RedirectExec.execute(RedirectExec.java:116)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.ExecChainElement.execute(ExecChainElement.java:51)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.InternalHttpClient.doExecute(InternalHttpClient.java:178)
        at com.github.dockerjava.zerodep.shaded.org.apache.hc.client5.http.impl.classic.CloseableHttpClient.execute(CloseableHttpClient.java:67)
        at com.github.dockerjava.zerodep.ApacheDockerHttpClientImpl.execute(ApacheDockerHttpClientImpl.java:206)
        at com.github.dockerjava.zerodep.ZerodepDockerHttpClient.execute(ZerodepDockerHttpClient.java:8)
        at com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:228)
        at com.github.dockerjava.core.DefaultInvocationBuilder.lambda$executeAndStream$1(DefaultInvocationBuilder.java:269)
        at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.UnsatisfiedLinkError: Error looking up function 'write': The specified procedure could not be found.
 [in thread "docker-java-stream--1108008968"]
        at com.sun.jna.Function.<init>(Function.java:255)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:618)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:594)
        at com.sun.jna.NativeLibrary.getFunction(NativeLibrary.java:580)
        at com.sun.jna.Native.register(Native.java:1900)
        at com.sun.jna.Native.register(Native.java:1771)
        at com.sun.jna.Native.register(Native.java:1489)
        at com.github.dockerjava.transport.DomainSocket.<clinit>(DomainSocket.java:54)
        ... 27 more
```

### Environment

- Kestra Version: 0.16.6
- Operating System: Windows Standalone JDBC backend
- Java Version `openjdk version "17.0.11" 2024-04-16`
- Installed plugins:
```
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a----          5/7/2024   4:06 PM       11825875 plugin-jdbc-postgres-0.16.1.jar
-a----          5/7/2024   4:06 PM         105692 plugin-script-powershell-0.16.1.jar
-a----          5/7/2024   4:06 PM         129964 plugin-script-python-0.16.1.jar
-a----          5/7/2024   4:06 PM       98931636 plugin-serdes-0.16.0.jar```

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java
@@ -8,7 +8,6 @@ import io.kestra.core.utils.Slugify;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.RandomStringUtils;
 import org.apache.commons.lang3.StringUtils;
-import org.apache.commons.lang3.SystemUtils;
 
 import javax.annotation.Nullable;
 import java.io.FileOutputStream;
@@ -103,11 +102,12 @@ public final class ScriptService {
         RunContext runContext,
         List<String> commands
     ) throws IOException, IllegalVariableEvaluationException {
-        return ScriptService.replaceInternalStorage(runContext, Collections.emptyMap(), commands, (ignored, file) -> {}, false);
+        return ScriptService.replaceInternalStorage(runContext, Collections.emptyMap(), commands, (ignored, file) -> {
+        }, false);
     }
 
     private static String saveOnLocalStorage(RunContext runContext, String uri) throws IOException {
-        try(InputStream inputStream = runContext.storage().getFile(URI.create(uri))) {
+        try (InputStream inputStream = runContext.storage().getFile(URI.create(uri))) {
             Path path = runContext.tempFile();
 
             IOUtils.copyLarge(inputStream, new FileOutputStream(path.toFile()));
@@ -137,11 +137,20 @@ public final class ScriptService {
         return uploaded;
     }
 
+
     public static List<String> scriptCommands(List<String> interpreter, List<String> beforeCommands, String command) {
-        return scriptCommands(interpreter, beforeCommands, List.of(command));
+        return scriptCommands(interpreter, beforeCommands, List.of(command), TargetOS.LINUX);
     }
 
     public static List<String> scriptCommands(List<String> interpreter, List<String> beforeCommands, List<String> commands) {
+        return scriptCommands(interpreter, beforeCommands, commands, TargetOS.LINUX);
+    }
+
+    public static List<String> scriptCommands(List<String> interpreter, List<String> beforeCommands, String command, TargetOS targetOS) {
+        return scriptCommands(interpreter, beforeCommands, List.of(command), targetOS);
+    }
+
+    public static List<String> scriptCommands(List<String> interpreter, List<String> beforeCommands, List<String> commands, TargetOS targetOS) {
         ArrayList<String> commandsArgs = new ArrayList<>(interpreter);
         commandsArgs.add(
             Stream
@@ -149,7 +158,7 @@ public final class ScriptService {
                     ListUtils.emptyOnNull(beforeCommands).stream(),
                     commands.stream()
                 )
-                .collect(Collectors.joining(getSeparator()))
+                .collect(Collectors.joining(targetOS.getLineSeparator()))
         );
 
         return commandsArgs;
@@ -237,12 +246,5 @@ public final class ScriptService {
         // we add a suffix of 5 chars, this should be enough as it's the standard k8s way
         String suffix = RandomStringUtils.randomAlphanumeric(5).toLowerCase();
         return normalized + "-" + suffix;
-    }
-
-    private static String getSeparator() {
-        if (SystemUtils.IS_OS_WINDOWS) {
-            return "\n";
-        }
-        return System.lineSeparator();
     }
 }

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/ScriptService.java
@@ -8,6 +8,7 @@ import io.kestra.core.utils.Slugify;
 import org.apache.commons.io.IOUtils;
 import org.apache.commons.lang3.RandomStringUtils;
 import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.SystemUtils;
 
 import javax.annotation.Nullable;
 import java.io.FileOutputStream;
@@ -148,7 +149,7 @@ public final class ScriptService {
                     ListUtils.emptyOnNull(beforeCommands).stream(),
                     commands.stream()
                 )
-                .collect(Collectors.joining(System.lineSeparator()))
+                .collect(Collectors.joining(getSeparator()))
         );
 
         return commandsArgs;
@@ -236,5 +237,12 @@ public final class ScriptService {
         // we add a suffix of 5 chars, this should be enough as it's the standard k8s way
         String suffix = RandomStringUtils.randomAlphanumeric(5).toLowerCase();
         return normalized + "-" + suffix;
+    }
+
+    private static String getSeparator() {
+        if (SystemUtils.IS_OS_WINDOWS) {
+            return "\n";
+        }
+        return System.lineSeparator();
     }
 }

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/TargetOS.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/TargetOS.java
@@ -1,0 +1,21 @@
+package io.kestra.core.models.tasks.runners;
+
+public enum TargetOS {
+    LINUX {
+        public String getLineSeparator() {
+            return "\n";
+        }
+    },
+    WINDOWS {
+        public String getLineSeparator() {
+            return "\r\n";
+        }
+    },
+    AUTO {
+        public String getLineSeparator() {
+            return System.lineSeparator();
+        }
+    };
+
+    public abstract String getLineSeparator();
+}

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/TaskCommands.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/TaskCommands.java
@@ -5,6 +5,7 @@ import java.time.Duration;
 import java.util.List;
 import java.util.Map;
 
+
 /**
  * Interface for the commands passed to a TaskRunner.
  */
@@ -30,4 +31,6 @@ public interface TaskCommands {
     }
 
     Duration getTimeout();
+
+    TargetOS getTargetOS();
 }

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java
@@ -6,6 +6,7 @@ import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.models.Plugin;
 import io.kestra.core.models.WorkerJobLifecycle;
 import io.kestra.core.runners.RunContext;
+import io.kestra.plugin.core.runner.Process;
 import jakarta.validation.constraints.NotBlank;
 import jakarta.validation.constraints.Pattern;
 import lombok.AccessLevel;
@@ -13,8 +14,11 @@ import lombok.Builder;
 import lombok.Getter;
 import lombok.NoArgsConstructor;
 import lombok.experimental.SuperBuilder;
+import org.apache.commons.lang3.SystemUtils;
 
-import java.util.*;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 
@@ -108,10 +112,19 @@ public abstract class TaskRunner implements Plugin, WorkerJobLifecycle {
         return new HashMap<>();
     }
 
-    public String toAbsolutePath(RunContext runContext, TaskCommands taskCommands, String relativePath) throws IllegalVariableEvaluationException {
+    public String toAbsolutePath(RunContext runContext, TaskCommands taskCommands, String relativePath, TargetOS targetOS) throws IllegalVariableEvaluationException {
         Object workingDir = this.additionalVars(runContext, taskCommands).get(ScriptService.VAR_WORKING_DIR);
         if (workingDir == null) {
+
             return relativePath;
+        }
+        // Case Target is Windows
+        if (targetOS.equals(TargetOS.WINDOWS) ||
+            // Case Target is AUTO and System is Windows while using Process runner
+            targetOS.equals(TargetOS.AUTO) && SystemUtils.IS_OS_WINDOWS && this instanceof Process
+        ) {
+
+            return (workingDir + "/" + relativePath).replace("\\", "/");
         }
 
         return windowsToUnixPath(workingDir + "/" + relativePath);

--- a/core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java
+++ b/core/src/main/java/io/kestra/core/models/tasks/runners/TaskRunner.java
@@ -18,6 +18,8 @@ import java.util.*;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicReference;
 
+import static io.kestra.core.utils.WindowsUtils.windowsToUnixPath;
+
 /**
  * Base class for all task runners.
  */
@@ -38,12 +40,12 @@ public abstract class TaskRunner implements Plugin, WorkerJobLifecycle {
     @JsonIgnore
     @Getter(AccessLevel.NONE)
     private transient Map<String, String> env;
-    
+
     @JsonIgnore
     @Builder.Default
     @Getter(AccessLevel.NONE)
     private AtomicReference<Runnable> killable = new AtomicReference<>();
-    
+
     @JsonIgnore
     @Builder.Default
     @Getter(AccessLevel.PROTECTED)
@@ -112,9 +114,9 @@ public abstract class TaskRunner implements Plugin, WorkerJobLifecycle {
             return relativePath;
         }
 
-        return workingDir + "/" + relativePath;
+        return windowsToUnixPath(workingDir + "/" + relativePath);
     }
-    
+
     /** {@inheritDoc} **/
     @Override
     public void kill() {
@@ -125,7 +127,7 @@ public abstract class TaskRunner implements Plugin, WorkerJobLifecycle {
             }
         }
     }
-    
+
     /**
      * Registers a runnable to be invoked when this {@link TaskRunner} is killed.
      * The passed {@link Runnable} can be used to dispose any resource or process started by the {@link TaskRunner}.

--- a/core/src/main/java/io/kestra/core/utils/WindowsUtils.java
+++ b/core/src/main/java/io/kestra/core/utils/WindowsUtils.java
@@ -1,9 +1,16 @@
 package io.kestra.core.utils;
 
+import java.util.regex.Matcher;
+
 public class WindowsUtils {
 
     public static String windowsToUnixPath(String path){
-        String unixPath = path.replace("\\", "/").replace(":", "");
+        Matcher matcher = java.util.regex.Pattern.compile("([A-Za-z]:)").matcher(path);
+        String unixPath = matcher.replaceAll(m -> m.group().toLowerCase());
+
+        unixPath = unixPath
+            .replace("\\", "/")
+            .replace(":", "");
         if (!unixPath.startsWith("/")) {
             unixPath = "/" + unixPath;
         }

--- a/core/src/main/java/io/kestra/core/utils/WindowsUtils.java
+++ b/core/src/main/java/io/kestra/core/utils/WindowsUtils.java
@@ -1,0 +1,12 @@
+package io.kestra.core.utils;
+
+public class WindowsUtils {
+
+    public static String windowsToUnixPath(String path){
+        String unixPath = path.replace("\\", "/").replace(":", "");
+        if (!unixPath.startsWith("/")) {
+            unixPath = "/" + unixPath;
+        }
+        return unixPath;
+    }
+}

--- a/core/src/test/java/io/kestra/core/models/tasks/runners/TaskRunnerTest.java
+++ b/core/src/test/java/io/kestra/core/models/tasks/runners/TaskRunnerTest.java
@@ -151,6 +151,11 @@ public class TaskRunnerTest {
         }
 
         @Override
+        public TargetOS getTargetOS() {
+            return null;
+        }
+
+        @Override
         public Map<String, Object> getAdditionalVars() {
             return Map.of(
                 SCRIPT_COMMANDS_ADDITIONAL_VAR_KEY, SCRIPT_COMMANDS_ADDITIONAL_VAR_VALUE,

--- a/script/src/main/java/io/kestra/plugin/scripts/exec/AbstractExecScript.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/exec/AbstractExecScript.java
@@ -4,6 +4,7 @@ import com.google.common.annotations.Beta;
 import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.models.annotations.PluginProperty;
 import io.kestra.core.models.tasks.*;
+import io.kestra.core.models.tasks.runners.TargetOS;
 import io.kestra.core.models.tasks.runners.TaskRunner;
 import io.kestra.core.runners.RunContext;
 import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
@@ -101,6 +102,12 @@ public abstract class AbstractExecScript extends Task implements RunnableTask<Sc
     @Deprecated
     private Boolean outputDirectory;
 
+    @Schema(
+        title = "The target operating system where the script will run."
+    )
+    @Builder.Default
+    public TargetOS targetOS = TargetOS.AUTO;
+
     abstract public DockerOptions getDocker();
 
     @Schema(
@@ -139,7 +146,8 @@ public abstract class AbstractExecScript extends Task implements RunnableTask<Sc
             .withInputFiles(this.inputFiles)
             .withOutputFiles(this.outputFiles)
             .withEnableOutputDirectory(this.getOutputDirectory())
-            .withTimeout(this.getTimeout());
+            .withTimeout(this.getTimeout())
+            .withTargetOS(this.targetOS);
     }
 
     protected List<String> getBeforeCommandsWithOptions() {

--- a/script/src/main/java/io/kestra/plugin/scripts/exec/AbstractExecScript.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/exec/AbstractExecScript.java
@@ -80,7 +80,8 @@ public abstract class AbstractExecScript extends Task implements RunnableTask<Sc
     @Builder.Default
     @Schema(
         title = "Fail the task on the first command with a non-zero status.",
-        description = "If set to `false` all commands will be executed one after the other. The final state of task execution is determined by the last command. Note that this property maybe be ignored if a non compatible interpreter is specified."
+        description = "If set to `false` all commands will be executed one after the other. The final state of task execution is determined by the last command. Note that this property maybe be ignored if a non compatible interpreter is specified." +
+            "\nYou can also disable it if your interpreter does not support the `set -e`option."
     )
     @PluginProperty
     protected Boolean failFast = true;

--- a/script/src/main/java/io/kestra/plugin/scripts/exec/scripts/runners/CommandsWrapper.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/exec/scripts/runners/CommandsWrapper.java
@@ -9,6 +9,7 @@ import io.kestra.core.runners.FilesService;
 import io.kestra.core.runners.NamespaceFilesService;
 import io.kestra.core.runners.RunContext;
 import io.kestra.core.utils.IdUtils;
+import io.kestra.plugin.scripts.exec.AbstractExecScript;
 import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
 import io.kestra.plugin.scripts.exec.scripts.models.RunnerType;
 import io.kestra.plugin.scripts.exec.scripts.models.ScriptOutput;
@@ -75,6 +76,9 @@ public class CommandsWrapper implements TaskCommands {
     @With
     private Duration timeout;
 
+    @With
+    private TargetOS targetOS;
+
     public CommandsWrapper(RunContext runContext) {
         this.runContext = runContext;
         this.workingDirectory = runContext.tempDir();
@@ -101,7 +105,8 @@ public class CommandsWrapper implements TaskCommands {
             inputFiles,
             outputFiles,
             enableOutputDirectory,
-            timeout
+            timeout,
+            targetOS
         );
     }
 

--- a/script/src/main/java/io/kestra/plugin/scripts/runner/docker/Docker.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/runner/docker/Docker.java
@@ -412,7 +412,7 @@ public class Docker extends TaskRunner {
         }
         boolean volumesEnabled = volumeEnabledConfig.orElse(Boolean.FALSE);
 
-        Path workingDirectory = Path.of(windowsToUnixPath(taskCommands.getWorkingDirectory().toString()));
+        Path workingDirectory = taskCommands.getWorkingDirectory();
         String image = runContext.render(this.image, additionalVars);
 
         CreateContainerCmd container = dockerClient.createContainerCmd(image);
@@ -428,7 +428,7 @@ public class Docker extends TaskRunner {
         );
 
         if (workingDirectory != null) {
-            container.withWorkingDir(windowsToUnixPath(workingDirectory.toFile().toString()));
+            container.withWorkingDir(windowsToUnixPath(workingDirectory.toAbsolutePath().toString()));
         }
 
         List<Bind> binds = new ArrayList<>();
@@ -521,6 +521,7 @@ public class Docker extends TaskRunner {
         if (this.getNetworkMode() != null) {
             hostConfig.withNetworkMode(runContext.render(this.getNetworkMode(), additionalVars));
         }
+
 
         return container
             .withHostConfig(hostConfig)

--- a/script/src/main/java/io/kestra/plugin/scripts/runner/docker/Docker.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/runner/docker/Docker.java
@@ -37,6 +37,7 @@ import java.util.stream.Collectors;
 
 import static io.kestra.core.utils.Rethrow.throwConsumer;
 import static io.kestra.core.utils.Rethrow.throwFunction;
+import static io.kestra.core.utils.WindowsUtils.windowsToUnixPath;
 
 @SuperBuilder
 @ToString
@@ -411,7 +412,7 @@ public class Docker extends TaskRunner {
         }
         boolean volumesEnabled = volumeEnabledConfig.orElse(Boolean.FALSE);
 
-        Path workingDirectory = taskCommands.getWorkingDirectory();
+        Path workingDirectory = Path.of(windowsToUnixPath(taskCommands.getWorkingDirectory().toString()));
         String image = runContext.render(this.image, additionalVars);
 
         CreateContainerCmd container = dockerClient.createContainerCmd(image);
@@ -427,15 +428,16 @@ public class Docker extends TaskRunner {
         );
 
         if (workingDirectory != null) {
-            container.withWorkingDir(workingDirectory.toFile().getAbsolutePath());
+            container.withWorkingDir(windowsToUnixPath(workingDirectory.toFile().toString()));
         }
 
         List<Bind> binds = new ArrayList<>();
 
         if (workingDirectory != null) {
+            String bindPath = windowsToUnixPath(workingDirectory.toString());
             binds.add(new Bind(
-                workingDirectory.toAbsolutePath().toString(),
-                new Volume(workingDirectory.toAbsolutePath().toString()),
+                bindPath,
+                new Volume(bindPath),
                 AccessMode.rw
             ));
         }

--- a/script/src/main/java/io/kestra/plugin/scripts/runner/docker/DockerService.java
+++ b/script/src/main/java/io/kestra/plugin/scripts/runner/docker/DockerService.java
@@ -10,6 +10,7 @@ import io.kestra.core.exceptions.IllegalVariableEvaluationException;
 import io.kestra.core.runners.RunContext;
 import io.kestra.core.serializers.JacksonMapper;
 import io.kestra.core.utils.MapUtils;
+import org.apache.commons.lang3.SystemUtils;
 
 import javax.annotation.Nullable;
 import java.io.File;
@@ -40,6 +41,10 @@ public class DockerService {
 
         if (Files.exists(Path.of("/var/run/docker.sock"))) {
             return "unix:///var/run/docker.sock";
+        }
+
+        if (SystemUtils.IS_OS_WINDOWS) {
+            return "npipe:////./pipe/docker_engine";
         }
 
         return "unix:///dind/docker.sock";

